pinot-minion/src/main/java/org/apache/pinot/minion/executor/BaseTaskExecutor.java

```diff
@@ -43,9 +44,20 @@ protected TableConfig getTableConfig(String tableNameWithType) {
   }
 
   protected Schema getSchema(String tableName) {
-    Schema schema =
-        ZKMetadataProvider.getTableSchema(MINION_CONTEXT.getHelixPropertyStore(), tableName);
+    Schema schema = ZKMetadataProvider.getTableSchema(MINION_CONTEXT.getHelixPropertyStore(), tableName);
     Preconditions.checkState(schema != null, "Failed to find schema for table: %s", tableName);
     return schema;
   }
+
+  /**
+   * Pre processing operations to be done at the beginning of task execution
+   */
+  protected void preProcess(PinotTaskConfig pinotTaskConfig) {
```

Comment:
Move these 2 methods to the `BaseMultipleSegmentsConversionExecutor` where it is getting called. Usually we allow child class to override methods used in the parent class.